engines/causal-chains/engine.js

```diff
@@ -0,0 +1,105 @@
+import * as fs from 'fs';
+import {execSync} from "child_process"
+import path from 'path';
+import {tmpdir} from 'os';
+import {fileURLToPath} from 'url';
+
+const __filename = fileURLToPath(import.meta.url); // get the resolved path to the file
+const __dirname = path.dirname(__filename); // get the name of the directory
+
+class Engine {
+    constructor() {
+    }
+
+    static supportedModes() {
+        // check that the ./causal-chains Go binary exists
+        try {
+            const stats = fs.statSync(`${__dirname}/causal-chains`);
+            const isExecutable = !!(stats.mode & (fs.constants.S_IXUSR | fs.constants.S_IXGRP | fs.constants.S_IXOTH));
+
+            if (isExecutable) {
+                return ["cld"];
```

Comment:
I tested this - if you delete the `causal-chains` binary and run `node evals/run.js -e ./evals/experiments/leaderboard.json`, you don't see the `causal-chains-o4-mini` scheduled to run, but if you then run `npm i` with Go installed (e.g. via `brew install go`), when you run `node evals/run.js -e ./evals/experiments/leaderboard.json` it will show up

Reply:
Confirmed